opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/AlarmRestService.java

```diff
@@ -0,0 +1,313 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v2;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.cxf.jaxrs.ext.search.SearchBean;
+import org.opennms.core.config.api.JaxbListWrapper;
+import org.opennms.core.criteria.Alias.JoinType;
+import org.opennms.core.criteria.CriteriaBuilder;
+import org.opennms.core.criteria.Fetch.FetchType;
+import org.opennms.core.criteria.restrictions.Restrictions;
+import org.opennms.core.resource.Vault;
+import org.opennms.netmgt.dao.api.AcknowledgmentDao;
+import org.opennms.netmgt.dao.api.AlarmDao;
+import org.opennms.netmgt.dao.api.AlarmRepository;
+import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.model.AckAction;
+import org.opennms.netmgt.model.OnmsAcknowledgment;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.netmgt.model.OnmsAlarmCollection;
+import org.opennms.netmgt.model.TroubleTicketState;
+import org.opennms.web.rest.support.Aliases;
+import org.opennms.web.rest.support.CriteriaBehavior;
+import org.opennms.web.rest.support.CriteriaBehaviors;
+import org.opennms.web.rest.support.IpLikeCriteriaBehavior;
+import org.opennms.web.rest.support.MultivaluedMapImpl;
+import org.opennms.web.rest.support.SecurityHelper;
+import org.opennms.web.svclayer.TroubleTicketProxy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Basic Web Service using REST for {@link OnmsAlarm} entity.
+ *
+ * @author <a href="agalue@opennms.org">Alejandro Galue</a>
+ */
+@Component
+@Path("alarms")
+@Transactional
+public class AlarmRestService extends AbstractDaoRestService<OnmsAlarm,SearchBean,Integer,Integer> {
+
+    @Autowired
+    private AlarmDao m_dao;
+
+    @Autowired
+    private AcknowledgmentDao m_ackDao;
+
+    @Autowired
+    private AlarmRepository m_repository;
+
+//    @Autowired
```

Comment:
The ticketer autowiring wasn't working during integration tests so I commented this field out. @mvrueden worked on this as part of HZN-1062. We'll need to get the tests working so that ticketing works properly.